db/wpdb.ts

```diff
@@ -660,6 +663,145 @@ export const getRelatedChartsForVariable = async (
             `)
 }
 
+interface RelatedResearchQueryResult {
+    linkTargetSlug: string
+    componentType: string
+    chartSlug: string
+    title: string
+    postSlug: string
+    chartId: number
+    authors: string
+    thumbnail: string
+    pageviews: number
+    post_source: string
+    tags: string
+}
+export const getRelatedResearchAndWritingForVariable = async (
+    variableId: number
+): Promise<DataPageRelatedResearch[]> => {
+    const wp_posts: RelatedResearchQueryResult[] = await db.queryMysql(
+        `-- sql
+            select
+                distinct
+                pl.target as linkTargetSlug,
+                pl.componentType as componentType,
+                c.slug as chartSlug,
+                p.title as title,
+                p.slug as postSlug,
+                coalesce(csr.chart_id, c.id) as chartId,
+                p.authors as authors,
+                p.featured_image as thumbnail,
+                coalesce(pv.views_365d, 0) as pageviews,
+                'wordpress' as post_source,
+                (select coalesce(JSON_ARRAYAGG(t.name), JSON_ARRAY())
+                    from post_tags pt
+                    join tags t on pt.tag_id = t.id
+                    where pt.post_id = p.id
+                ) as tags
+            from
+                posts_links pl
+            join posts p on
+                pl.sourceId = p.id
+            left join charts c on
+                pl.target = c.slug
+            left join chart_slug_redirects csr on
+                pl.target = csr.slug
+            left join chart_dimensions cd on
+                cd.chartId = c.id
+            left join pageviews pv on
+                pv.url = concat('https://ourworldindata.org/', p.slug )
+            left join posts_gdocs pg on
+            	pg.id = p.gdocSuccessorId
+            left join post_tags pt on
+                pt.post_id = p.id
+            where
+                pl.linkType = 'grapher'
+                and componentType = 'src' -- this filters out links in tags and keeps only embedded charts
```

Comment:
I don't know what that comment means? Can we reword it somehow?

Reply:
(explaining for my own understanding as well 😊)

`linkType = 'grapher'` is for URLs with paths that match on `/^\/grapher\/[\w]+/`

`componentType = 'src'` is for the links that match the `anySrcRegex` (and not the `anyHrefRegex` or `prominentLinkRegex`

So this clause is filtering out any links that may be from an inline link to a grapher, because we only want to count grapher embeds (i.e. links from iframes - `<iframe src="https://ourworldindata.org/grapher/life-expectancy"`)